src/internal/operations/maintenance.go

```diff
@@ -0,0 +1,76 @@
+package operations
+
+import (
+	"context"
+	"time"
+
+	"github.com/alcionai/clues"
+
+	"github.com/alcionai/corso/src/internal/common/crash"
+	"github.com/alcionai/corso/src/internal/events"
+	"github.com/alcionai/corso/src/internal/kopia"
+	"github.com/alcionai/corso/src/internal/stats"
+	"github.com/alcionai/corso/src/pkg/control"
+)
+
+// MaintenanceOperation wraps an operation with restore-specific props.
+type MaintenanceOperation struct {
+	operation
+	Results          MaintenanceResults
+	safety           control.Safety
+	force            bool
+	quickMaintenance bool
+}
+
+// MaintenanceResults aggregate the details of the results of the operation.
+type MaintenanceResults struct {
+	stats.StartAndEndTime
+}
+
+// NewMaintenanceOperation constructs and validates a restore operation.
+func NewMaintenanceOperation(
+	ctx context.Context,
+	opts control.Options,
+	kw *kopia.Wrapper,
+	safety control.Safety,
+	quickMaintenance, force bool,
```

Comment:
is having `quickMaintenace` be a bool sufficiently generic? Should we change it to an enum instead so that we can add more values in the future if we need to?

This is going to be the SDK-level API so it would be best to try to make it fit our future needs if we can

Reply:
Enums over flags are always a good idea, imo.